drivers/usb/device/usb_dc_mcux_ehci.c

```diff
@@ -37,7 +37,7 @@ extern void USB_DeviceEhciIsrFunction(void *deviceHandle);
 /* Then endpoint number/index calculation */
 #define EP_ABS_IDX(ep)		(USB_EP_GET_IDX(ep) * 2 + \
 					(USB_EP_GET_DIR(ep) >> 7))
-#define NUM_OF_EP_MAX		DT_INST_PROP(0, num_bidir_endpoints)
+#define NUM_OF_EP_MAX		(DT_INST_PROP(0, num_bidir_endpoints) * 2)
```

Comment:
I understand what you are doing here. One observation is that this NUM_OF_EP_MAX is the total number of endpoints but each endpoint is not equal. As long as the reader of the code understands that, at least in this case, this is the total count with an equal split between IN and OUT endpoints.

I also think that your comments in the commit should be pulled into the source code somewhere for the viewer to see and understand.